frontend/lib/src/components/graph.dart

```diff
@@ -39,95 +40,112 @@ class GraphState extends State {
           ],
         ),
         padding: const EdgeInsets.all(20),
-        child: SizedBox(
-          width: double.maxFinite,
-          height: 236,
-          child: Row(
-            mainAxisAlignment: MainAxisAlignment.spaceEvenly,
-            children: [
-              SizedBox(
-                width: 180,
-                height: 180,
-                child: PieChart(
-                  PieChartData(
-                    pieTouchData: PieTouchData(
-                      touchCallback: (FlTouchEvent event, pieTouchResponse) {
-                        setState(() {
-                          if (!event.isInterestedForInteractions ||
-                              pieTouchResponse == null ||
-                              pieTouchResponse.touchedSection == null) {
-                            focusIndex = -1;
-                            return;
-                          }
-                          focusIndex = pieTouchResponse
-                              .touchedSection!.touchedSectionIndex;
-                        });
-                      },
+        child: InkWell(
+          onTap: onTap,
+          child: SizedBox(
+            width: double.maxFinite,
+            height: 236,
+            child: Row(
+              mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+              children: [
+                SizedBox(
+                  width: 150,
+                  height: 150,
+                  child: StatefulBuilder(
+                    builder: (context, stateful) {
+                      return PieChart(
+                        PieChartData(
+                          pieTouchData: PieTouchData(
+                            touchCallback:
+                                (FlTouchEvent event, pieTouchResponse) {
+                              stateful(() {
+                                if (!event.isInterestedForInteractions ||
+                                    pieTouchResponse == null ||
+                                    pieTouchResponse.touchedSection == null) {
+                                  controller.setFocusIndex = -1;
+
+                                  return;
+                                }
+                                controller.setFocusIndex = pieTouchResponse
+                                    .touchedSection!.touchedSectionIndex;
+                              });
+                            },
+                          ),
+                          borderData: FlBorderData(
+                            show: false,
+                          ),
+                          sectionsSpace: 0,
+                          centerSpaceRadius: 30,
+                          sections: showingSections(),
```

Comment:
no need for this to be a function right?

Reply:
it's needed, i can't use controllers states if I don't make it a function.